File: app/src/main/java/org/michaelbel/tjgram/data/room/UserDao.java
package org.michaelbel.tjgram.data.room;

import java.util.List;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

@Dao
public interface UserDao {

    @Query("SELECT * FROM users")
    List<User> getAll();

    @Query("SELECT * FROM users WHERE id = :id")
    User getById(int id);

    @Insert
    void insert(User user);

    @Update
    void update(User user);

    @Delete
    void delete(User user);
}

File: app/src/main/java/org/michaelbel/tjgram/data/persistence/dao/UserDao.kt
package org.michaelbel.tjgram.data.persistence.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import io.reactivex.Completable
import io.reactivex.Flowable
import org.michaelbel.tjgram.data.persistence.entities.LocalUser

@Dao
interface UserDao {

    @Query("SELECT * FROM Users WHERE id = :id")
    fun getUserById(id: Int): Flowable<LocalUser>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertUser(user: LocalUser): Completable

    @Query("DELETE FROM Users")
    fun deleteAllUsers()
}